Updating integrations asset reference with manifest v2 information

anawishnoff opened the pull request:
### What does this PR do?
<!-- A brief description of the change being made with this pull request. -->
Adding tabs and a new table to this doc that provide information on the attributes of manifest v2

### Motivation
<!-- What inspired you to submit this pull request? -->
The current doc was outdated

### Additional Notes
<!-- Anything else we should know when reviewing? -->

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have `changelog/` and `integration/` labels attached

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `tile[media[media_type]]`                            | String or Enum                                                          | Mandatory                                  | The type of media this is. If you're setting this field as an Enum, the only currently allowed values for the Enum are “image” and “video”.                                                                                     ...
+| `tile[media[caption]]`                               | String                                                                  | Mandatory                                  | The caption for the image                                                                                                                                                                                                       ...
+| `tile[media[image_url]]`                             | String                                                                  | Mandatory                                  | The relative path to this image in relation to this manifest file                                                                                                                                                               ...
+| `tile[description]`                                  | String\[80\]                                                            | Mandatory                                  | A brief description of what this offering provides. Limited to 80 characters                                                                                                                                                    ...
```
```suggestion
| `tile[description]`                                  | String\[80\]                                                            | Mandatory                                  | A brief description of what this offering provides. Limited to 80 characters.                                                                                                                                                                                   |   || `tile[title]`                                        | String\[50\]                                                            | Mandatory                                  | The user-friendly title for this app.                                                                                                                                                                                                                      |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `tile[media]`                                        | Array of Object                                                         | Mandatory, can be `[ ]`                    | Information about various image/video style objects that will be presented in the media gallery carousel on the listing page.                                                                                                   ...
+| `tile[media[media_type]]`                            | String or Enum                                                          | Mandatory                                  | The type of media this is. If you're setting this field as an Enum, the only currently allowed values for the Enum are “image” and “video”.                                                                                     ...
+| `tile[media[caption]]`                               | String                                                                  | Mandatory                                  | The caption for the image                                                                                                                                                                                                       ...
+| `tile[media[image_url]]`                             | String                                                                  | Mandatory                                  | The relative path to this image in relation to this manifest file                                                                                                                                                               ...
```
```suggestion
| `tile[media[image_url]]`                             | String                                                                  | Mandatory                                  | The relative path to this image in relation to this manifest file.                                                                                                                                                                                         |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `pricing`                                            | Array of Object                                                         | Mandatory                                  | List of objects representing the pricing model of the integration. See [Marketplace GitHub repository][1] for pricing details. Note that the Marketplace GitHub repository is private - please email marketplace@datadog.com wi ...
+| `tile[media]`                                        | Array of Object                                                         | Mandatory, can be `[ ]`                    | Information about various image/video style objects that will be presented in the media gallery carousel on the listing page.                                                                                                   ...
+| `tile[media[media_type]]`                            | String or Enum                                                          | Mandatory                                  | The type of media this is. If you're setting this field as an Enum, the only currently allowed values for the Enum are “image” and “video”.                                                                                     ...
+| `tile[media[caption]]`                               | String                                                                  | Mandatory                                  | The caption for the image                                                                                                                                                                                                       ...
```
```suggestion
| `tile[media[caption]]`                               | String                                                                  | Mandatory                                  | The caption for the image.                                                                                                                                                                                                                                 |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `manifest_version`                                   | String Enum                                                             | Mandatory    | Version of the manifest schema. Supported values: `1.0.0` and `2.0.0`                                                                                                                                                                                     |   ...
+| `pricing`                                            | Array of Object                                                         | Mandatory                                  | List of objects representing the pricing model of the integration. See [Marketplace GitHub repository][1] for pricing details. Note that the Marketplace GitHub repository is private - please email marketplace@datadog.com wi ...
+| `tile[media]`                                        | Array of Object                                                         | Mandatory, can be `[ ]`                    | Information about various image/video style objects that will be presented in the media gallery carousel on the listing page.                                                                                                   ...
+| `tile[media[media_type]]`                            | String or Enum                                                          | Mandatory                                  | The type of media this is. If you're setting this field as an Enum, the only currently allowed values for the Enum are “image” and “video”.                                                                                     ...
```
```suggestion
| `tile[media[media_type]]`                            | String or Enum                                                          | Mandatory                                  | The type of media this is. If you're setting this field as an Enum, only `image` and `video` are accepted.                                                                                                               |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `legal_terms[eula] `                                 | String                                                                  | Mandatory                                  | Relative path to the EULA (End User License Agreement) PDF in relation to this manifest                                                                                                                                         ...
+| `manifest_version`                                   | String Enum                                                             | Mandatory    | Version of the manifest schema. Supported values: `1.0.0` and `2.0.0`                                                                                                                                                                                     |   ...
+| `pricing`                                            | Array of Object                                                         | Mandatory                                  | List of objects representing the pricing model of the integration. See [Marketplace GitHub repository][1] for pricing details. Note that the Marketplace GitHub repository is private - please email marketplace@datadog.com wi ...
+| `tile[media]`                                        | Array of Object                                                         | Mandatory, can be `[ ]`                    | Information about various image/video style objects that will be presented in the media gallery carousel on the listing page.                                                                                                   ...
```
```suggestion
| `tile[media]`                                        | Array of Object                                                         | Mandatory, can be `[ ]`                    | Information about various image and video style objects that are presented in the media gallery carousel on the listing page.                                                                                                                             |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `legal_terms `                                       | Object                                                                  | Mandatory                                  | Any legal documentation that needs to be agreed to by a user to use this app.                                                                                                                                                   ...
+| `legal_terms[eula] `                                 | String                                                                  | Mandatory                                  | Relative path to the EULA (End User License Agreement) PDF in relation to this manifest                                                                                                                                         ...
+| `manifest_version`                                   | String Enum                                                             | Mandatory    | Version of the manifest schema. Supported values: `1.0.0` and `2.0.0`                                                                                                                                                                                     |   ...
+| `pricing`                                            | Array of Object                                                         | Mandatory                                  | List of objects representing the pricing model of the integration. See [Marketplace GitHub repository][1] for pricing details. Note that the Marketplace GitHub repository is private - please email marketplace@datadog.com wi ...
```
```suggestion
| `pricing`                                            | Array of Object                                                         | Mandatory                                  | List of objects representing the pricing model of the integration. See [Marketplace GitHub repository][1] for pricing details. The Marketplace GitHub repository is private, email marketplace@datadog.com for access.                                                                                                                                                                                          |   || `tile`                                               | Object                                                                  | Mandatory                                  | Information about this offering                                                                                                                                                                                                                                |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `display_on_public_website `                         | Boolean                                                                 | Mandatory                 | Whether information about this listing should be displayed on the public Datadog docs site. Once this is set to True, it cannot be changed.                                                                                                      ...
+| `legal_terms `                                       | Object                                                                  | Mandatory                                  | Any legal documentation that needs to be agreed to by a user to use this app.                                                                                                                                                   ...
+| `legal_terms[eula] `                                 | String                                                                  | Mandatory                                  | Relative path to the EULA (End User License Agreement) PDF in relation to this manifest                                                                                                                                         ...
+| `manifest_version`                                   | String Enum                                                             | Mandatory    | Version of the manifest schema. Supported values: `1.0.0` and `2.0.0`                                                                                                                                                                                     |   ...
```
```suggestion
| `manifest_version`                                   | String Enum                                                             | Mandatory    | Version of the manifest schema. Supported values include `1.0.0` and `2.0.0`.                                                                                                                                                                                     |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `classifier_tags`                                    | Array of String                                                         | Mandatory, can be `[ ]`                    | Some classifier information about this App. This can include information such as the `supported_os` and `available_offerings`.                                                                                                  ...
+| `display_on_public_website `                         | Boolean                                                                 | Mandatory                 | Whether information about this listing should be displayed on the public Datadog docs site. Once this is set to True, it cannot be changed.                                                                                                      ...
+| `legal_terms `                                       | Object                                                                  | Mandatory                                  | Any legal documentation that needs to be agreed to by a user to use this app.                                                                                                                                                   ...
+| `legal_terms[eula] `                                 | String                                                                  | Mandatory                                  | Relative path to the EULA (End User License Agreement) PDF in relation to this manifest                                                                                                                                         ...
```
```suggestion
| `legal_terms[eula] `                                 | String                                                                  | Mandatory                                  | Relative path to the EULA (End User License Agreement) PDF in relation to this manifest.                                                                                                                                                                   |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `author[vendor_id]`                                  | String                                                                  | Mandatory                                  | The id of the vendor to use for subscription purposes. Must be globally unique and cannot be changed. Should follow the strict standards of `integration_id` where only dashes and alphabetic chars are allowed. This value wil ...
+| `classifier_tags`                                    | Array of String                                                         | Mandatory, can be `[ ]`                    | Some classifier information about this App. This can include information such as the `supported_os` and `available_offerings`.                                                                                                  ...
+| `display_on_public_website `                         | Boolean                                                                 | Mandatory                 | Whether information about this listing should be displayed on the public Datadog docs site. Once this is set to True, it cannot be changed.                                                                                                      ...
+| `legal_terms `                                       | Object                                                                  | Mandatory                                  | Any legal documentation that needs to be agreed to by a user to use this app.                                                                                                                                                   ...
```
```suggestion
| `legal_terms `                                       | Object                                                                  | Mandatory                                  | Any legal documentation that a user needs to agree to in order to use this app.                                                                                                                                                                             |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `author[support_email]`                              | String (Email)                                                          | Mandatory                                  | The email to contact for any support/maintenance queries                                                                                                                                                                        ...
+| `author[vendor_id]`                                  | String                                                                  | Mandatory                                  | The id of the vendor to use for subscription purposes. Must be globally unique and cannot be changed. Should follow the strict standards of `integration_id` where only dashes and alphabetic chars are allowed. This value wil ...
+| `classifier_tags`                                    | Array of String                                                         | Mandatory, can be `[ ]`                    | Some classifier information about this App. This can include information such as the `supported_os` and `available_offerings`.                                                                                                  ...
+| `display_on_public_website `                         | Boolean                                                                 | Mandatory                 | Whether information about this listing should be displayed on the public Datadog docs site. Once this is set to True, it cannot be changed.                                                                                                      ...
```
```suggestion
| `display_on_public_website `                         | Boolean                                                                 | Mandatory                 | Whether or not information about this listing is displayed on the public Datadog docs site. Once this is set to True, it cannot be changed.                                                                                                               |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `author[sales_email]`                                | String (Email)                                                          | Mandatory                                  | The email to contact for any subscription level events                                                                                                                                                                          ...
+| `author[support_email]`                              | String (Email)                                                          | Mandatory                                  | The email to contact for any support/maintenance queries                                                                                                                                                                        ...
+| `author[vendor_id]`                                  | String                                                                  | Mandatory                                  | The id of the vendor to use for subscription purposes. Must be globally unique and cannot be changed. Should follow the strict standards of `integration_id` where only dashes and alphabetic chars are allowed. This value wil ...
+| `classifier_tags`                                    | Array of String                                                         | Mandatory, can be `[ ]`                    | Some classifier information about this App. This can include information such as the `supported_os` and `available_offerings`.                                                                                                  ...
```
```suggestion
| `classifier_tags`                                    | Array of String                                                         | Mandatory, can be `[ ]`                    | Some classifier information about this app. This includes information such as `supported_os` and `available_offerings`.                                                                                                                            |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `author[name]`                                       | String                                                                  | Mandatory                                  | The human readable name for this company                                                                                                                                                                                        ...
+| `author[sales_email]`                                | String (Email)                                                          | Mandatory                                  | The email to contact for any subscription level events                                                                                                                                                                          ...
+| `author[support_email]`                              | String (Email)                                                          | Mandatory                                  | The email to contact for any support/maintenance queries                                                                                                                                                                        ...
+| `author[vendor_id]`                                  | String                                                                  | Mandatory                                  | The id of the vendor to use for subscription purposes. Must be globally unique and cannot be changed. Should follow the strict standards of `integration_id` where only dashes and alphabetic chars are allowed. This value wil ...
```
```suggestion
| `author[vendor_id]`                                  | String                                                                  | Mandatory                                  | The vendor ID to use for subscription purposes. Must be globally unique and cannot be changed. Should follow the strict standards of `integration_id` where only dashes and alphabetic chars are allowed. This value is provided to partners. |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `author `                                            | Object                                                                  | Mandatory                                  | Information about the author of this App                                                                                                                                                                                        ...
+| `author[homepage] `                                  | String (URL)                                                            | Mandatory                                  | The web URL to the homepage of the author                                                                                                                                                                                       ...
+| `author[name]`                                       | String                                                                  | Mandatory                                  | The human readable name for this company                                                                                                                                                                                        ...
+| `author[sales_email]`                                | String (Email)                                                          | Mandatory                                  | The email to contact for any subscription level events                                                                                                                                                                          ...
```
```suggestion
| `author[sales_email]`                                | String (Email)                                                          | Mandatory                                  | The email to contact for any subscription-level events.                                                                                                                                                                                                    |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `author[homepage] `                                  | String (URL)                                                            | Mandatory                                  | The web URL to the homepage of the author                                                                                                                                                                                       ...
+| `author[name]`                                       | String                                                                  | Mandatory                                  | The human readable name for this company                                                                                                                                                                                        ...
+| `author[sales_email]`                                | String (Email)                                                          | Mandatory                                  | The email to contact for any subscription level events                                                                                                                                                                          ...
+| `author[support_email]`                              | String (Email)                                                          | Mandatory                                  | The email to contact for any support/maintenance queries                                                                                                                                                                        ...
```
```suggestion
| `author[support_email]`                              | String (Email)                                                          | Mandatory                                  | The email to contact for any support and maintenance queries.                                                                                                                                                                                                  |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[monitors["monitor_short_name"]]`             |                                                                         |                                           | Key/value pairs for any recommended monitors. The key is the globally unique short name of the dashboard and the value is the relative path to the monitors JSON definition in relation to this manifest                         ...
+| `author `                                            | Object                                                                  | Mandatory                                  | Information about the author of this App                                                                                                                                                                                        ...
+| `author[homepage] `                                  | String (URL)                                                            | Mandatory                                  | The web URL to the homepage of the author                                                                                                                                                                                       ...
+| `author[name]`                                       | String                                                                  | Mandatory                                  | The human readable name for this company                                                                                                                                                                                        ...
```
```suggestion
| `author[name]`                                       | String                                                                  | Mandatory                                  | The human-readable name for this company.                                                                                                                                                                                                                  |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[monitors]`                                   | Object                                                                  | Mandatory, can be `{ }`                    | Recommended monitors                                                                                                                                                                                                            ...
+| `assets[monitors["monitor_short_name"]]`             |                                                                         |                                           | Key/value pairs for any recommended monitors. The key is the globally unique short name of the dashboard and the value is the relative path to the monitors JSON definition in relation to this manifest                         ...
+| `author `                                            | Object                                                                  | Mandatory                                  | Information about the author of this App                                                                                                                                                                                        ...
+| `author[homepage] `                                  | String (URL)                                                            | Mandatory                                  | The web URL to the homepage of the author                                                                                                                                                                                       ...
```
```suggestion
| `author[homepage] `                                  | String (URL)                                                            | Mandatory                                  | The web URL to the homepage of the author.                                                                                                                                                                                                                 |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[source_type-name]]`              | String                                                                  | Mandatory                                  | Same as `display_name`                                                                                                                                                                                                          ...
+| `assets[monitors]`                                   | Object                                                                  | Mandatory, can be `{ }`                    | Recommended monitors                                                                                                                                                                                                            ...
+| `assets[monitors["monitor_short_name"]]`             |                                                                         |                                           | Key/value pairs for any recommended monitors. The key is the globally unique short name of the dashboard and the value is the relative path to the monitors JSON definition in relation to this manifest                         ...
+| `author `                                            | Object                                                                  | Mandatory                                  | Information about the author of this App                                                                                                                                                                                        ...
```
```suggestion
| `author `                                            | Object                                                                  | Mandatory                                  | Information about the author of this app.                                                                                                                                                                                                                  |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[supported_os]]`                  | List of String (Enum)                                                   | Mandatory, can be `[ ]`                    | The available operating systems this integration can be used on                                                                                                                                                                 ...
+| `assets[integration[source_type-name]]`              | String                                                                  | Mandatory                                  | Same as `display_name`                                                                                                                                                                                                          ...
+| `assets[monitors]`                                   | Object                                                                  | Mandatory, can be `{ }`                    | Recommended monitors                                                                                                                                                                                                            ...
+| `assets[monitors["monitor_short_name"]]`             |                                                                         |                                           | Key/value pairs for any recommended monitors. The key is the globally unique short name of the dashboard and the value is the relative path to the monitors JSON definition in relation to this manifest                         ...
```
```suggestion
| `assets[monitors["monitor_short_name"]]`             |                                                                         |                                           | Key/value pairs for any recommended monitors. The key is the globally unique short name of the dashboard and the value is the relative path to the monitors JSON definition in relation to this manifest.                                                  |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[metrics[prefix]]]`               | String                                                                  | Mandatory                                  | The prefix for metrics emitted by this integration                                                                                                                                                                              ...
+| `assets[integration[service_checks]]`                | Object                                                                  | Mandatory, but can be `{ }`                | Information about service checks emitted by this integration                                                                                                                                                                    ...
+| `assets[integration[service_checks[metadata_path]]]` | String                                                                  | Mandatory                                  | Relative path to where the service check metadata lives in relation to this manifest                                                                                                                                            ...
+| `assets[integration[supported_os]]`                  | List of String (Enum)                                                   | Mandatory, can be `[ ]`                    | The available operating systems this integration can be used on                                                                                                                                                                 ...
```
```suggestion
| `assets[integration[supported_os]]`                  | List of String (Enum)                                                   | Mandatory, can be `[ ]`                    | The available operating systems this integration can be used on.                                                                                                                                                                                           |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[service_checks]]`                | Object                                                                  | Mandatory, but can be `{ }`                | Information about service checks emitted by this integration                                                                                                                                                                    ...
+| `assets[integration[service_checks[metadata_path]]]` | String                                                                  | Mandatory                                  | Relative path to where the service check metadata lives in relation to this manifest                                                                                                                                            ...
+| `assets[integration[supported_os]]`                  | List of String (Enum)                                                   | Mandatory, can be `[ ]`                    | The available operating systems this integration can be used on                                                                                                                                                                 ...
+| `assets[integration[source_type-name]]`              | String                                                                  | Mandatory                                  | Same as `display_name`                                                                                                                                                                                                          ...
```
```suggestion
| `assets[integration[source_type-name]]`              | String                                                                  | Mandatory                                  | Same as `display_name`.                                                                                                                                                                                                                                    |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[service_checks[metadata_path]]]` | String                                                                  | Mandatory                                  | Relative path to where the service check metadata lives in relation to this manifest                                                                                                                                            ...
+| `assets[integration[supported_os]]`                  | List of String (Enum)                                                   | Mandatory, can be `[ ]`                    | The available operating systems this integration can be used on                                                                                                                                                                 ...
+| `assets[integration[source_type-name]]`              | String                                                                  | Mandatory                                  | Same as `display_name`                                                                                                                                                                                                          ...
+| `assets[monitors]`                                   | Object                                                                  | Mandatory, can be `{ }`                    | Recommended monitors                                                                                                                                                                                                            ...
```
```suggestion
| `assets[monitors]`                                   | Object                                                                  | Mandatory, can be `{ }`                    | Recommended monitors.                                                                                                                                                                                                                                      |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[metrics[auto_install]]]`         | Boolean                                                                 | Mandatory                                  | Boolean indicating if this metric should automatically be marked as installed if the `assets[integration[metrics[check]]]` comes in                                                                                             ...
+| `assets[integration[metrics[check]]]`                | String or List of String                                                | Mandatory                                  | A string or list representing a metric that this integration always emits on every run                                                                                                                                          ...
+| `assets[integration[metrics[metadata_path]]]`        | String                                                                  | Mandatory                                  | Relative path to where the metrics metadata lives in relation to this manifest                                                                                                                                                  ...
+| `assets[integration[metrics[prefix]]]`               | String                                                                  | Mandatory                                  | The prefix for metrics emitted by this integration                                                                                                                                                                              ...
```
```suggestion
| `assets[integration[metrics[prefix]]]`               | String                                                                  | Mandatory                                  | The prefix for metrics emitted by this integration.                                                                                                                                                                                                        |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[metrics[check]]]`                | String or List of String                                                | Mandatory                                  | A string or list representing a metric that this integration always emits on every run                                                                                                                                          ...
+| `assets[integration[metrics[metadata_path]]]`        | String                                                                  | Mandatory                                  | Relative path to where the metrics metadata lives in relation to this manifest                                                                                                                                                  ...
+| `assets[integration[metrics[prefix]]]`               | String                                                                  | Mandatory                                  | The prefix for metrics emitted by this integration                                                                                                                                                                              ...
+| `assets[integration[service_checks]]`                | Object                                                                  | Mandatory, but can be `{ }`                | Information about service checks emitted by this integration                                                                                                                                                                    ...
```
```suggestion
| `assets[integration[service_checks]]`                | Object                                                                  | Mandatory, but can be `{ }`                | Information about service checks emitted by this integration.                                                                                                                                                                                              |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[metrics[metadata_path]]]`        | String                                                                  | Mandatory                                  | Relative path to where the metrics metadata lives in relation to this manifest                                                                                                                                                  ...
+| `assets[integration[metrics[prefix]]]`               | String                                                                  | Mandatory                                  | The prefix for metrics emitted by this integration                                                                                                                                                                              ...
+| `assets[integration[service_checks]]`                | Object                                                                  | Mandatory, but can be `{ }`                | Information about service checks emitted by this integration                                                                                                                                                                    ...
+| `assets[integration[service_checks[metadata_path]]]` | String                                                                  | Mandatory                                  | Relative path to where the service check metadata lives in relation to this manifest                                                                                                                                            ...
```
```suggestion
| `assets[integration[service_checks[metadata_path]]]` | String                                                                  | Mandatory                                  | Relative path to where the service check metadata lives in relation to this manifest.                                                                                                                                                                      |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[metrics]]`                       | Object                                                                  | Mandatory                                  | Information about the metrics this integration emits                                                                                                                                                                            ...
+| `assets[integration[metrics[auto_install]]]`         | Boolean                                                                 | Mandatory                                  | Boolean indicating if this metric should automatically be marked as installed if the `assets[integration[metrics[check]]]` comes in                                                                                             ...
+| `assets[integration[metrics[check]]]`                | String or List of String                                                | Mandatory                                  | A string or list representing a metric that this integration always emits on every run                                                                                                                                          ...
+| `assets[integration[metrics[metadata_path]]]`        | String                                                                  | Mandatory                                  | Relative path to where the metrics metadata lives in relation to this manifest                                                                                                                                                  ...
```
```suggestion
| `assets[integration[metrics[metadata_path]]]`        | String                                                                  | Mandatory                                  | Relative path to where the metrics metadata lives in relation to this manifest.                                                                                                                                                                            |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[events[creates_events]]]`        | Boolean                                                                 | Mandatory                  | Whether or not this integration emits events to Datadog                                                                                                                                                                                         ...
+| `assets[integration[metrics]]`                       | Object                                                                  | Mandatory                                  | Information about the metrics this integration emits                                                                                                                                                                            ...
+| `assets[integration[metrics[auto_install]]]`         | Boolean                                                                 | Mandatory                                  | Boolean indicating if this metric should automatically be marked as installed if the `assets[integration[metrics[check]]]` comes in                                                                                             ...
+| `assets[integration[metrics[check]]]`                | String or List of String                                                | Mandatory                                  | A string or list representing a metric that this integration always emits on every run                                                                                                                                          ...
```
```suggestion
| `assets[integration[metrics[check]]]`                | String or List of String                                                | Mandatory                                  | A string or list representing a metric that this integration always emits on every run.                                                                                                                                                                    |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[configuration[spec]]]`           | String                                                                  | Mandatory                                  | Relative path to where the configuration spec lives in relation to this manifest                                                                                                                                                ...
+| `assets[integration[events]]`                        | Object                                                                  | Mandatory                                  | Information about events emitted by this integration                                                                                                                                                                            ...
+| `assets[integration[events[creates_events]]]`        | Boolean                                                                 | Mandatory                  | Whether or not this integration emits events to Datadog                                                                                                                                                                                         ...
+| `assets[integration[metrics]]`                       | Object                                                                  | Mandatory                                  | Information about the metrics this integration emits                                                                                                                                                                            ...
```
```suggestion
| `assets[integration[metrics]]`                       | Object                                                                  | Mandatory                                  | Information about the metrics this integration emits.                                                                                                                                                                                                      |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| Attribute | Type  | Mandatory/Optional       | Description                                                                                                                                                                                                                                               |   |
+|------------------------------------------------------|-------------------------------------------------------------------------|-------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| `app_id`                                             | String                                                                  | Mandatory                                  | Globally unique name of the offering. Should follow the same standards and limitations of `integration_id` as it is defined today.               |   |
+| `app_uuid`                                           | UUID                                                                    | Mandatory                                  | Globally unique UUID for this Application.                                                                                                                                                   |   |
```
```suggestion
| `app_uuid`                                           | UUID                                                                    | Mandatory                                  | Globally unique UUID for this application.                                                                                                                                                   |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `app_id`                                             | String                                                                  | Mandatory                                  | Globally unique name of the offering. Should follow the same standards and limitations of `integration_id` as it is defined today.               |   |
+| `app_uuid`                                           | UUID                                                                    | Mandatory                                  | Globally unique UUID for this Application.                                                                                                                                                   |   |
+| `assets`                                             | Object                                                                  | Mandatory                                  | Object containing any Datadog installable entity.                                                                                                |   |
+| `assets[dashboards]`                                 | Object                                                                  | Mandatory, but can be `{ }`                | Out-of-the-box  dashboards associated with this offering                                                                                                                                                                        ...
```
```suggestion
| `assets[dashboards]`                                 | Object                                                                  | Mandatory, but can be `{ }`                | Out-of-the-box dashboards associated with this offering.                                                                                                                                                                                                       |   |
```

alai97 commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets`                                             | Object                                                                  | Mandatory                                  | Object containing any Datadog installable entity.                                                                                                |   |
+| `assets[dashboards]`                                 | Object                                                                  | Mandatory, but can be `{ }`                | Out-of-the-box  dashboards associated with this offering                                                                                                                                                                        ...
+| `assets[dashboards["dashboard_short_name"]]`         | String                                                                  | Mandatory                                  | Key/value pairs of any out-of-the-box dashboards. The key is the globally unique short name of the dashboard and the value is the relative path to the dashboard's JSON definition in relation to this manifest.                ...
+| `assets[integration]`                                | Object                                                                  | Mandatory, but can be `{ }`                | Object containing information about the integration                                                                                                                                                                             ...
```
```suggestion
| `assets[integration]`                                | Object                                                                  | Mandatory, but can be `{ }`                | Object containing information about the integration.                                                                                                                                                                                                       |   |
```

anawishnoff commented:
Thanks for the in-depth review @alai97! Just addressed all of your comments and waiting on the Marketplace product/engineering team to review and approve as well.

nmuesch commented on docs/dev/check_references.md:
```diff
@@ -99,6 +111,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets`                                             | Object                                                                  | Mandatory                                  | Object containing any Datadog installable entity.                                                                                                |   |
+| `assets[dashboards]`                                 | Object                                                                  | Mandatory, but can be `{ }`                | Out-of-the-box dashboards associated with this offering.                                                                                                                                                                        ...
+| `assets[dashboards["dashboard_short_name"]]`         | String                                                                  | Mandatory                                  | Key/value pairs of any out-of-the-box dashboards. The key is the globally unique short name of the dashboard and the value is the relative path to the dashboard's JSON definition in relation to this manifest.                ...
+| `assets[integration]`                                | Object                                                                  | Mandatory, but can be `{ }`                | Object containing information about the integration.                                                                                                                                                                            ...
```
All root "objects" under "assets" is actually `Optional` instead of `{}`

Ex 
* assets->dashboards
* assets->integration
are both `optional`. 

Anything under those keys though are good as they are (i.e.. `assets->integration->configuration`)

nmuesch commented on docs/dev/check_references.md:
```diff
@@ -99,6 +111,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[metrics[prefix]]]`               | String                                                                  | Mandatory                                  | The prefix for metrics emitted by this integration.                                                                                                                                                                             ...
+| `assets[integration[service_checks]]`                | Object                                                                  | Mandatory, but can be `{ }`                | Information about service checks emitted by this integration.                                                                                                                                                                   ...
+| `assets[integration[service_checks[metadata_path]]]` | String                                                                  | Mandatory                                  | Relative path to where the service check metadata lives in relation to this manifest.                                                                                                                                           ...
+| `assets[integration[supported_os]]`                  | List of String (Enum)                                                   | Mandatory, can be `[ ]`                    | The available operating systems this integration can be used on.                                                                                                                                                                ...
```
For manifest V2, supported_os is now part of the `classifier_tags`. This row can be removed

nmuesch commented on docs/dev/check_references.md:
```diff
@@ -99,6 +111,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[service_checks]]`                | Object                                                                  | Mandatory, but can be `{ }`                | Information about service checks emitted by this integration.                                                                                                                                                                   ...
+| `assets[integration[service_checks[metadata_path]]]` | String                                                                  | Mandatory                                  | Relative path to where the service check metadata lives in relation to this manifest.                                                                                                                                           ...
+| `assets[integration[supported_os]]`                  | List of String (Enum)                                                   | Mandatory, can be `[ ]`                    | The available operating systems this integration can be used on.                                                                                                                                                                ...
+| `assets[integration[source_type-name]]`              | String                                                                  | Mandatory                                  | Same as `display_name`.                                                                                                                                                                                                         ...
```
```suggestion
| `assets[integration[source_type_name]]`              | String                                                                  | Mandatory                                  | Same as `display_name`.                                                                                                                                                                                                                                    |   |
```

nmuesch commented on docs/dev/check_references.md:
```diff
@@ -99,6 +111,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `assets[integration[service_checks[metadata_path]]]` | String                                                                  | Mandatory                                  | Relative path to where the service check metadata lives in relation to this manifest.                                                                                                                                           ...
+| `assets[integration[supported_os]]`                  | List of String (Enum)                                                   | Mandatory, can be `[ ]`                    | The available operating systems this integration can be used on.                                                                                                                                                                ...
+| `assets[integration[source_type-name]]`              | String                                                                  | Mandatory                                  | Same as `display_name`.                                                                                                                                                                                                         ...
+| `assets[monitors]`                                   | Object                                                                  | Mandatory, can be `{ }`                    | Recommended monitors.                                                                                                                                                                                                           ...
```
```suggestion
| `assets[monitors]`                                   | Object                                                                  | Optional                   | Recommended monitors.                                                                                                                                                                                                                                      |   |
```

nmuesch commented on docs/dev/check_references.md:
```diff
@@ -99,6 +111,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `author[name]`                                       | String                                                                  | Mandatory                                  | The human-readable name for this company.                                                                                                                                                                                       ...
+| `author[sales_email]`                                | String (Email)                                                          | Mandatory                                  | The email to contact for any subscription-level events.                                                                                                                                                                         ...
+| `author[support_email]`                              | String (Email)                                                          | Mandatory                                  | The email to contact for any support and maintenance queries.                                                                                                                                                                   ...
+| `author[vendor_id]`                                  | String                                                                  | Mandatory                                  | The vendor ID to use for subscription purposes. Must be globally unique and cannot be changed. Should follow the strict standards of `integration_id` where only dashes and alphabetic chars are allowed. This value is provide ...
```
Lets change this to reference app_id since V2 doesn't have an `integration_id`

```suggestion
| `author[vendor_id]`                                  | String                                                                  | Mandatory                                  | The vendor ID to use for subscription purposes. Must be globally unique and cannot be changed. Should follow the strict standards of `app_id` where only dashes and alphabetic chars are allowed. This value is provided to partners. |   |
```

nmuesch commented on docs/dev/check_references.md:
```diff
@@ -99,6 +111,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `author[support_email]`                              | String (Email)                                                          | Mandatory                                  | The email to contact for any support and maintenance queries.                                                                                                                                                                   ...
+| `author[vendor_id]`                                  | String                                                                  | Mandatory                                  | The vendor ID to use for subscription purposes. Must be globally unique and cannot be changed. Should follow the strict standards of `integration_id` where only dashes and alphabetic chars are allowed. This value is provide ...
+| `classifier_tags`                                    | Array of String                                                         | Mandatory, can be `[ ]`                    | Some classifier information about this app. This includes information such as `supported_os` and `available_offerings`.                                                                                                         ...
+| `display_on_public_website `                         | Boolean                                                                 | Mandatory                 | Whether or not information about this listing is displayed on the public Datadog docs site. Once this is set to True, it cannot be changed.                                                                                                      ...
```
We should confirm with someone on the docs team if the "cannot be changed" constraint is still valid
@davidejones would you be able to confirm if this is still a constraint?

nmuesch commented on docs/dev/check_references.md:
```diff
@@ -99,6 +111,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `display_on_public_website `                         | Boolean                                                                 | Mandatory                 | Whether or not information about this listing is displayed on the public Datadog docs site. Once this is set to True, it cannot be changed.                                                                                                      ...
+| `legal_terms `                                       | Object                                                                  | Mandatory                                  | Any legal documentation that a user needs to agree to in order to use this app.                                                                                                                                                 ...
+| `legal_terms[eula] `                                 | String                                                                  | Mandatory                                  | Relative path to the EULA (End User License Agreement) PDF in relation to this manifest.                                                                                                                                        ...
+| `manifest_version`                                   | String Enum                                                             | Mandatory    | Version of the manifest schema. Supported values include `1.0.0` and `2.0.0`.                                                                                                                                                                                 ...
```
Nit, can this be the first entry of this table?

nmuesch commented on docs/dev/check_references.md:
```diff
@@ -99,6 +111,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
+| `manifest_version`                                   | String Enum                                                             | Mandatory    | Version of the manifest schema. Supported values include `1.0.0` and `2.0.0`.                                                                                                                                                                                 ...
+| `pricing`                                            | Array of Object                                                         | Mandatory                                  | List of objects representing the pricing model of the integration. See [Marketplace GitHub repository][1] for pricing details. The Marketplace GitHub repository is private, email marketplace@datadog.com for access.          ...
+| `tile[media]`                                        | Array of Object                                                         | Mandatory, can be `[ ]`                    | Information about various image and video style objects that are presented in the media gallery carousel on the listing page.                                                                                                   ...
+| `tile[media[media_type]]`                            | String or Enum                                                          | Mandatory                                  | The type of media this is. If you're setting this field as an Enum, only `image` and `video` are accepted.                                                                                                               |   |
```
We've implemented this as an enum, so to a user the only allowed values are `video` and `image`. Suggestion to update the wording to be a bit clearer there:

```suggestion
| `tile[media[media_type]]`                            | String or Enum                                                          | Mandatory                                  | The type of media this is. Allowed values are `image` and `video`.                                                                                                               |   |
```

nmuesch reviewed (changes requested):
Thanks for this, left some inline comments/suggestions

anawishnoff commented on docs/dev/check_references.md:
```diff
@@ -99,6 +109,76 @@ The complete list of mandatory and optional attributes for the `manifest.json` f
 | `assets`-> `dashboards`     | Dictionary      | Mandatory          | Dictionary where the key is the name of the dashboard (must be globally unique across integrations) and the value is the relative file path where the dashboard definition lives.                                        |
 | `assets`-> `monitors`       | Dictionary      | Mandatory          | Dictionary where the key is the name of the monitor (must be globally unique across integrations) and the value is the relative file path where the dashboard definition lives.                                          |
 | `assets`-> `service_checks` | String          | Mandatory          | Relative location of where the `service_checks.json` file lives.                                                                                                                                                         |
+[1]: https://docs.datadoghq.com/integrations/
+[2]: https://www.uuidgenerator.net
```
I think for tabs specifically numbered links are supposed to go at the bottom of the tab

nmuesch replied:
Sounds good, thanks for confirming!

nmuesch reviewed (dismissed):
Content LGTM, there's just the one outstanding comment about the documentation field.

alexvetras commented on docs/dev/check_references.md:
```diff
@@ -71,15 +71,72 @@ For more information about YAML syntax, see [Wikipedia][2]. Feel free to play ar
+| Attribute | Type  | Mandatory/Optional       | Description                                                                                                                                                                                                                                               |   |
+|------------------------------------------------------|-------------------------------------------------------------------------|-------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| `manifest_version`                                   | String Enum                                                             | Mandatory    | Version of the manifest schema. Supported values include `1.0.0` and `2.0.0`.                                                                                                                                                                                 ...
+| `app_id`                                             | String                                                                  | Mandatory                                  | The unique identifying name of this offering. Usually kebab case of the display name.              |   |
```
Small question/suggestion: 

Does it make sense to actually write in "kebab-case," or add an example for clarification? 

With the JSON object name being in snake_case, but needing the identifying name to be in kebab-case, seems like it could be helpful.

anawishnoff replied:
Added an example in upcoming commit

alexvetras commented on docs/dev/check_references.md:
```diff
@@ -71,15 +71,72 @@ For more information about YAML syntax, see [Wikipedia][2]. Feel free to play ar
+|------------------------------------------------------|-------------------------------------------------------------------------|-------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| `manifest_version`                                   | String Enum                                                             | Mandatory    | Version of the manifest schema. Supported values include `1.0.0` and `2.0.0`.                                                                                                                                                                                 ...
+| `app_id`                                             | String                                                                  | Mandatory                                  | The unique identifying name of this offering. Usually kebab case of the display name.              |   |
+| `app_uuid`                                           | UUID                                                                    | Mandatory                                  | Globally unique UUID for this application.                                                                                                                                                   |   |
```
Is it worth calling out the differences/need for both app_id and app_uuid?

anawishnoff replied:
Going to address this in a future PR once we get more info

alexvetras reviewed (approved):
LGTM!

anawishnoff commented:
I think this is good to merge whenever agent integrations/docs can give it a thumbs up :)
